src/picklescan/scanner.py

```diff
@@ -179,7 +178,17 @@ def _list_globals(data: IO[bytes], multiple_pickles=True) -> Set[Tuple[str, str]
         try:
             ops = list(pickletools.genops(data))
         except Exception as e:
-            raise GenOpsError(str(e))
+            # XXX: pickle will happily load files that contain arbitrarily placed new lines whereas pickletools errors in such cases.
+            # below is code to circumvent or skip these newlines while succeeding at parsing the opcodes.
+            err = str(e)
+            if "opcode b'\\n' unknown" not in err:
+                raise GenOpsError(err)
+            else:
+                pos = int(err.split(",")[0].replace("at position ", ""))
```

Comment:
Could you add a test for this branch? Parsing exception message strings is indeed pretty flaky and it would be good to get a heads up when this regresses (ex: future version of Python). It looks like data.pkl in the offending PyTorch file is only 16KB, so could be included in tests data if that's OK. Alternatively, malicious11.pkl is handcrafted in the test code and a variation could be added with extra new lines.
https://github.com/mmaitre314/picklescan/blob/12a1353aa4e8667c49de4e468947d2abc2994da3/tests/test_scanner.py#L219